src/controllers/hid/hidcontroller.cpp

```diff
@@ -245,23 +245,31 @@ bool HidController::poll() {
     Trace hidRead("HidController poll");
 
     int result = 1;
+    int bufferSize = sizeof(m_pPollData[0]) / sizeof(m_pPollData[0][0]);
     while (result > 0) {
-        result = hid_read(m_pHidDevice, m_pPollData, sizeof(m_pPollData) / sizeof(m_pPollData[0]));
+        // Rotate between two buffers so the memcmp below does not require deep copying to another buffer.
+        unsigned char* pCurrentBuffer = m_pPollData[m_iPollingBufferIndex];
+        if (m_iPollingBufferIndex == 0) {
+            m_iPollingBufferIndex = 1;
+        } else {
+            m_iPollingBufferIndex = 0;
+        }
+        unsigned char* pPreviousBuffer = m_pPollData[m_iPollingBufferIndex];
+
+        result = hid_read(m_pHidDevice, pCurrentBuffer, bufferSize);
         if (result == -1) {
             return false;
         } else if (result > 0) {
             Trace process("HidController process packet");
-            auto byteArray = QByteArray::fromRawData(reinterpret_cast<char*>(m_pPollData), result);
+            auto byteArray = QByteArray::fromRawData(
+                    reinterpret_cast<char*>(pCurrentBuffer), result);
             // Some controllers such as the Gemini GMX continuously send input packets even if it
             // is identical to the previous packet. If this loop processed all those redundant
             // packets, it would be a big performance problem.
             // TODO: Test if this hack can be done with adequate performance in JS so a deep copy
```

Comment:
This TODO is not necessary anymore, right?

Reply:
This still has not been tested.

Reply:
But I thought we don't do a deep copy anymore?

Reply:
Oh, you're right. I'll delete the comment.

Reply:
done in 7cdc090